DESIGN.md

```diff
@@ -0,0 +1,107 @@
+# ODD
+
+## Introduction
+
+The ODD SDK is a local-first toolkit that helps developers build systems that store data in a secure, reliable, and decentralised way. The focus is on the user, providing them with sufficient authority while still being convenient.
+
+ODD stands for Open Distributed Data. We want our data accessible from anywhere and distributed, meaning that it can live partially or fully in multiple places. Ideally as much as possible of this data lives on the devices of the owner of the data and communication of this data would be done in a decentralised peer-to-peer fashion.
+
+We also prefer not to rely on centralised services when providing users and devices with the needed authority. When a user starts using a new device, their existing device should be authorise the new one without any intervention from a central service. This can be done by giving the new device one or more tickets that can be verified on the spot.
+
+Rather than putting everything in a closed system, ODD puts everything out in the open. That means encryption is used to keep data private. These encryption keys and authorisation tickets give us a chance to escape from passwords, so preferably passwords are avoided.
+
+The ODD SDK relies on open protocols, this includes the [WNFS protocol](https://github.com/wnfs-wg) for the data management, and [UCAN](https://github.com/ucan-wg) to provide users and devices with the necessary authorisation.
+
+## Implementation
+
+ts-odd is written in Typescript, a Javascript dialect. The resulting code should be usable in any modern Javascript environment that supports the [Web Crypto API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API) and [Web Assembly](https://webassembly.org/). It uses the Rust implementations of [WNFS](https://github.com/wnfs-wg/rs-wnfs) and [UCAN](https://github.com/ucan-wg/rs-ucan) which compile to Web Assembly.
```

Comment:
Small tweak: `which` must proceeded by a comma in this context. Alternatively, it could be switched out for `that` with no comma before it
```suggestion
ts-odd is written in Typescript, a Javascript dialect. The resulting code should be usable in any modern Javascript environment that supports the [Web Crypto API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API) and [Web Assembly](https://webassembly.org/). It uses the Rust implementations of [WNFS](https://github.com/wnfs-wg/rs-wnfs) and [UCAN](https://github.com/ucan-wg/rs-ucan), which compile to Web Assembly.
```